backend/file-orchestrator-service/src/jvmMain/kotlin/service/SyncFolderService.kt

```diff
@@ -0,0 +1,320 @@
+                    (it.getLong("resource") ?: 0).toString(),
+                    SyncFolder.Spec(
+                        it.getString("path").orEmpty(),
+                        ProductReference(
+                            (it.getLong("product") ?: 0).toString(),
+                            it.getString("category").orEmpty(),
+                            it.getString("provider").orEmpty(),
+                        )
+                    ),
+                    0L,
+                    SyncFolder.Status(),
+                    emptyList(),
+                    ResourceOwner(
+                        it.getString("created_by").orEmpty(),
+                        null
+                    ),
+                    ResourcePermissions(emptyList(), emptyList())
+                )
+            }
+
+            if (affectedFolders.length > 0) {
+                proxy.bulkProxy(
+                    ActorAndProject(Actor.System, null),
+                    bulkRequestOf(affectedFolders.map { it.id }),
+                    BulkProxyInstructions.pureProcedure(
+                        service = this,
+                        retrieveCall = { providerApi(it).delete },
+                        requestToId = { it },
+                        resourceToRequest = { req, res -> res },
+                        verifyRequest = { _, _, _ -> {} }
+                    )
+                )
+
+                session.sendPreparedStatement(
+                    {
+                        setParameter("ids", affectedFolders.map { it.id })
+                    },
+                    """
+                            delete from file_orchestrator.sync_folders
+                            where resource in (select unnest(:ids::bigint[]))
+                        """
+                )
+
+                session.sendPreparedStatement(
+                    {
+                        setParameter("ids", affectedFolders.map { it.id })
+                    },
+                    """
+                            delete from provider.resource_acl_entry
+                            where resource_id in (select unnest(:ids::bigint[]))
+                        """
+                )
+
+                session.sendPreparedStatement(
+                    {
+                        setParameter("ids", affectedFolders.map { it.id })
+                    },
+                    """               
+                            delete from provider.resource_update
+                            where resource in (select unnest(:ids::bigint[]))
+                        """
+                )
+
+                session.sendPreparedStatement(
+                    {
+                        setParameter("ids", affectedFolders.map { it.id })
+                    },
+                    """
+                            delete from provider.resource
+                            where id in (select unnest(:ids::bigint[]))
+                        """
+                )
```

Comment:
We might want to consider a function or similar to do this. We are paying quite a lot of overhead for each query.